### PR TITLE
Disable all LocalCache Tests for Windows

### LINKED ISSUE #307: :farmer: 7 failing tests in Windows due to colon in `common::joinPaths`
## Environment
* OS Version: 
  * Windows
* Source build
  * Fuel tools 8

## Description
* Expected behavior: Build passing successful
* Actual behavior: Build has 7 test regressions

## Steps to reproduce
1. Run a build in [ign_fuel-tools-gz-8-win](https://build.osrfoundation.org/job/ign_fuel-tools-gz-8-win/)
2. See 7 test regressions when you finish

### Fuel tools 8 Windows | Test regressions | Pending investigation :yellow_circle: 

Reference build: https://build.osrfoundation.org/job/ign_fuel-tools-gz-8-win/19/#showFailuresLink

Failing tests:
<details>

* [UNIT_gz_src_TEST.test_ran](https://build.osrfoundation.org/job/ign_fuel-tools-gz-8-win/19/testReport/junit/(root)/UNIT_gz_src_TEST/test_ran/)
* [FuelClientTest.CachedWorld](https://build.osrfoundation.org/job/ign_fuel-tools-gz-8-win/19/testReport/junit/(root)/FuelClientTest/CachedWorld/)
* [FuelClientTest.PatchModelFail](https://build.osrfoundation.org/job/ign_fuel-tools-gz-8-win/19/testReport/junit/(root)/FuelClientTest/PatchModelFail/)
* [LocalCacheTest.MatchingModels](https://build.osrfoundation.org/job/ign_fuel-tools-gz-8-win/19/testReport/junit/(root)/LocalCacheTest/MatchingModels/)
* [LocalCacheTest.MatchingModel](https://build.osrfoundation.org/job/ign_fuel-tools-gz-8-win/19/testReport/junit/(root)/LocalCacheTest/MatchingModel/)
* [LocalCacheTest.AllWorlds](https://build.osrfoundation.org/job/ign_fuel-tools-gz-8-win/19/testReport/junit/(root)/LocalCacheTest/AllWorlds/)
* [LocalCacheTest.MatchingWorlds](https://build.osrfoundation.org/job/ign_fuel-tools-gz-8-win/19/testReport/junit/(root)/LocalCacheTest/MatchingWorlds/)
* [LocalCacheTest.MatchingWorld](https://build.osrfoundation.org/job/ign_fuel-tools-gz-8-win/19/testReport/junit/(root)/LocalCacheTest/MatchingWorld/)

</details>

---

FuelClientTest output:
```
C:\Jenkins\workspace\ign_fuel-tools-gz-8-win\ws\gz-fuel-tools\src\FuelClient_TEST.cc:108
Value of: common::createDirectories(common::joinPaths(worldPath, "2"))
  Actual: false
Expected: true
```

LocalCacheTest output:
```
C:\Jenkins\workspace\ign_fuel-tools-gz-8-win\ws\gz-fuel-tools\src\LocalCache_TEST.cc:137
Value of: common::createDirectories(common::joinPaths(serverPath, "alice", "worlds", "am1", "2"))
  Actual: false
Expected: true
```

This test is probably failing because of [Windows doesn't support colons in filenames](https://github.com/gazebosim/gz-fuel-tools/issues/106) (reported 2 years ago).

Colon is used in this [line of code](https://github.com/gazebosim/gz-fuel-tools/blob/gz-fuel-tools8/src/FuelClient_TEST.cc#L106), and the test is failing 2 lines below:

```cpp
  auto worldPath = common::joinPaths(
      "test_cache", "localhost:8007", "banana", "worlds", "My World");

  ASSERT_TRUE(common::createDirectories(common::joinPaths(worldPath, "2")));
  ASSERT_TRUE(common::createDirectories(common::joinPaths(worldPath, "3")));
```

### PR DESCRIPTION
**Pull request by Crola1702**
# 🦟 Bug fix

Fixes #307 (Partially)

## Summary
This PR disables LocalCache Tests for Windows

## Checklist
- [X] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by azeey — APPROVED** [review]

**Comment by Crola1702** [comment]
Test failures are known issues (Check https://github.com/gazebosim/gz-fuel-tools/issues/340)

**Review by Blast545 — APPROVED** [review]
LGTM, take a closer look to CI tests results before merging

**Comment by Crola1702** [comment]
@osrf-jenkins retest this please

**Comment by Crola1702** [comment]
As I mentioned above, test regressions in Windows CI are caused by https://github.com/gazebosim/gz-fuel-tools/issues/340.

@azeey do I get your double approval to merge this with these known test regressions making windows CI fail?

**Comment by azeey** [comment]
Yes, please go ahead

**Pull request merged**
